tests/test_tasks.py

```diff
@@ -1948,6 +1951,195 @@ def my_flow():
         futures = my_flow()
         assert [future.result() for future in futures] == [2, 3, 4]
 
+    @task
+    def echo(x):
+        return x
+
+    @task
+    def numbers():
+        return [1, 2, 3]
+
+    async def get_dependency_ids(self, session, flow_run_id) -> Dict[UUID, List[UUID]]:
+        graph = await models.flow_runs.read_task_run_dependencies(
+            session=session, flow_run_id=flow_run_id
+        )
+
+        return {x["id"]: [d.id for d in x["upstream_dependencies"]] for x in graph}
+
+    async def test_map_preserves_dependencies_between_futures_all_mapped_children(
+        self, session
+    ):
+        """
```

Comment:
I love these docstrings, such a great way of showing the complexity involved while giving a better understanding of the test. We should consider doing this in other places, some tests are pretty hard to decode.

Reply:
A very small nitpick: Without more of a description of the _test_ these should probably be standard comments instead of docstrings.

Reply:
Thank you! I was drawing these to better understand the relationship so thought why not include them for future readers. Will update to standard comments.

Reply:
Updated to docstrings of the flow as that is what they are a description of